### PR TITLE
Improve coverage

### PR DESCRIPTION
**Pull request by marcelm**
- Add a couple of tests
- Add a couple of `# pragma: no coverage`
- Remove some old code we no longer use

### CODE REVIEW

**Review by rhpvorderman — COMMENTED** [review]
Great. I love it when code gets deleted. Since 3.6's os.fspath is universally available now I spotted some small ways to delete even more code.

File: src/dnaio/singleend.py [hunk 1]
```diff
@@ -75,11 +61,36 @@ def _open_single(
...
         f"File format '{fileformat}' is unknown (expected 'fasta' or 'fastq')."
     )
 
 
+def _open_file_or_path(
+    file_or_path: Union[str, os.PathLike, BinaryIO], mode: str, opener
+) -> Tuple[bool, BinaryIO, str]:
+    path: Optional[str]
+    file: BinaryIO
+    if _is_path(file_or_path):
```
[the remark this answers is not in the document]
Comment (marcelm) [reply src/dnaio/singleend.py]: Ok, `_is_path` is gone (but at the cost of a `try:`/`else:`).

[the remark this answers is not in the document]
Comment (rhpvorderman) [reply src/dnaio/singleend.py]: Somehow this comment was not delivered to my e-mail address. Sorry for reacting so late.

**Comment by rhpvorderman** [comment]
Looks good to me!

**Comment by marcelm** [comment]
Thanks!

**Pull request merged**